prismatic/training/strategies/fsdp.py

```diff
@@ -175,7 +175,11 @@ def check_fn(submodule: nn.Module) -> bool:
                 return isinstance(submodule, self.llm_transformer_layer_cls)
 
             # Note that the terms "activation checkpointing" and "gradient checkpointing" are synonymous!
-            apply_activation_checkpointing(self.vlm, checkpoint_wrapper_fn=non_reentrant_wrapper, check_fn=check_fn)
+            if self.vlm.llm_backbone.llm_family == "open_lm":
+                self.vlm.llm_backbone.enable_gradient_checkpointing()
```

Comment:
why do we need this if/else compared to other vlms?

Reply:
I'm not sure why the checkpointing strategy from prismatic seems to not be compatible with OpenLM... I was getting errors:
```
prismatic/lib/python3.10/site-packages/torch/utils/checkpoint.py", line 853, in check_recomputed_tensors_match
    raise CheckpointError(
torch.utils.checkpoint.CheckpointError: torch.utils.checkpoint: A different number of tensors was saved during the original forward and recomputation.
Number of tensors saved during forward: 26
Number of tensors saved during recomputation: 5
```

Reply:
i see. we probably dont need gradient checkpointing anyways for training but we can figure that out later